--- npu_utils.cpp
+++ npu_utils_new.cpp
@@ -316,37 +316,6 @@
 	return Py_None;
 }

-static const std::unordered_map<std::string, rtLimitType_t> LimitTypeMap = {
-	{"LOW_POWER_TIMEOUT", rtLimitType_t::RT_LIMIT_TYPE_LOW_POWER_TIMEOUT},
-	{"WARP_STACK_SIZE", rtLimitType_t::RT_LIMIT_TYPE_SIMT_WARP_STACK_SIZE},
-	{"DVG_WARP_STACK_SIZE", rtLimitType_t::RT_LIMIT_TYPE_SIMT_DVG_WARP_STACK_SIZE},
-	{"STACK_SIZE", rtLimitType_t::RT_LIMIT_TYPE_STACK_SIZE}
-  };
-
-  static PyObject *setDeviceLimit(PyObject *self, PyObject *args) {
-	int device;              // device ID
-	const char *type_str;
-	uint32_t val;
-	if (!PyArg_ParseTuple(args, "isI", &device, &type_str, &val)) {
-	  return NULL;
-	}
-
-	auto it = LimitTypeMap.find(type_str);
-	if (it == LimitTypeMap.end()) {
-	  printf("Invalid limit type: %s.\n", type_str);
-	  return NULL;
-	}
-
-	rtError_t rtRet = rtDeviceSetLimit(device, it->second, val);
-	if (rtRet != RT_ERROR_NONE) {
-	  printf("rtDeviceSetLimit failed, 0x%x\n", rtRet);
-	  return NULL;
-	}
-	if (PyErr_Occurred()) {
-	  return NULL;
-	}
-	return Py_None;
-  }

 static PyMethodDef NpuUtilsMethods[] = {
     {"load_kernel_binary", loadKernelBinary, METH_VARARGS,
@@ -360,7 +329,6 @@
 	{"allocate_device_memory", allocateDeviceMemory, METH_VARARGS, "Allocate device memory"},
 	{"allocate_host_memory", allocateHostMemory, METH_VARARGS, "Allocate host memory"},
 	{"copy_memory", copyMemory, METH_VARARGS, "Copy data between host and device"},
-	{"set_device_limit", setDeviceLimit, METH_VARARGS, "Set the limit of NPU"},
     {NULL, NULL, 0, NULL}};

 static PyModuleDef ModuleDef = {
