--- models/qwen3_moe/modeling_qwen3_moe.py
+++ models/qwen3_moe/modeling_qwen3_moe.py
@@ -246,8 +246,9 @@
         expert_mask = torch.nn.functional.one_hot(selected_experts, num_classes=self.num_experts).permute(2, 1, 0)

         # Loop over all available experts in the model and perform the computation on each expert
-        expert_hit = torch.greater(expert_mask.sum(dim=(-1, -2)), 0).nonzero()
-        for expert_idx in expert_hit:
+        #expert_hit = torch.greater(expert_mask.sum(dim=(-1, -2)), 0).nonzero()
+        #for expert_idx in expert_hit:
+        for expert_idx in range(self.num_experts):
             expert_layer = self.experts[expert_idx]
             idx, top_x = torch.where(expert_mask[expert_idx].squeeze(0))
